src/vs/workbench/browser/parts/editor/textDiffEditor.ts

```diff
@@ -240,18 +240,17 @@ export class TextDiffEditor extends AbstractTextEditor<IDiffEditorViewState> imp
 		return {
 			...super.getConfigurationOverrides(),
 			readOnly,
-			dropIntoEditor: { enabled: !readOnly },
 			originalEditable: this.input instanceof DiffEditorInput && !this.input.original.hasCapability(EditorInputCapabilities.Readonly),
 			lineDecorationsWidth: '2ch'
 		};
 	}
 
 	protected override updateReadonly(input: EditorInput): void {
 		if (input instanceof DiffEditorInput) {
+			const readonly = input.hasCapability(EditorInputCapabilities.Readonly);
 			this.diffEditorControl?.updateOptions({
-				readOnly: input.hasCapability(EditorInputCapabilities.Readonly),
+				readOnly: readonly,
```

Comment:
could rename `readonly` -> `readOnly` 

then line 252 could be simplified to `readOnly`